datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py

```diff
@@ -226,48 +264,70 @@ def _submit_metrics(self, metrics, metric_name_and_type_by_property):
 
+    def get_running_wmi_sampler(self, properties, filters, **kwargs):
+        # type: (Iterable[str], List[Dict[str, WMIFilter]], Dict) -> WMISampler
+        return self._get_running_wmi_sampler(
+            instance_key=None,
+            wmi_class=self.wmi_class,
+            properties=properties,
+            tag_by=self.tag_by,
+            filters=filters,
+            host=self.host,
+            namespace=self.namespace,
+            provider=self.provider,
+            username=self.username,
+            password=self.password,
+            **kwargs
+        )
+
     def _get_running_wmi_sampler(self, instance_key, wmi_class, properties, tag_by="", **kwargs):
+        # type: (Any, str, Iterable[str], str, Any) -> WMISampler
         """
         Return a running WMISampler for the given (class, properties).
 
         If no matching WMISampler is running yet, start one and cache it.
         """
-        properties = list(properties) + [tag_by] if tag_by else list(properties)
+        if not self._wmi_sampler:
+            properties = list(properties) + [tag_by] if tag_by else list(properties)
+            self._wmi_sampler = WMISampler(self.log, wmi_class, properties, **kwargs)
+            self._wmi_sampler.start()
 
-        if instance_key not in self.wmi_samplers:
-            wmi_sampler = WMISampler(self.log, wmi_class, properties, **kwargs)
-            wmi_sampler.start()
-            self.wmi_samplers[instance_key] = wmi_sampler
+        return self._wmi_sampler
 
-        return self.wmi_samplers[instance_key]
+    def get_wmi_properties(self):
+        # type: () -> WMIProperties
+        return self._get_wmi_properties(None, self.metrics_to_capture, self.tag_queries)
 
     def _get_wmi_properties(self, instance_key, metrics, tag_queries):
+        # type: (Any, List[List[str]], List[List[str]]) -> WMIProperties
         """
         Create and cache a (metric name, metric type) by WMI property map and a property list.
         """
-        if instance_key not in self.wmi_props:
+        if not self._wmi_props:
             metric_name_by_property = dict(
                 (wmi_property.lower(), (metric_name, metric_type)) for wmi_property, metric_name, metric_type in metrics
-            )
-            properties = map(lambda x: x[0], metrics + tag_queries)
-            self.wmi_props[instance_key] = (metric_name_by_property, properties)
+            )  # type: Dict[str, Tuple[str, str]]
+            properties = map(lambda x: x[0], metrics + tag_queries)  # type: Iterable[str]
+
+            self._wmi_props = (metric_name_by_property, properties)
 
-        return self.wmi_props[instance_key]
+        return self._wmi_props
 
 
 def from_time(
     year=None, month=None, day=None, hours=None, minutes=None, seconds=None, microseconds=None, timezone=None
 ):
+    # type: (Optional[int], Optional[int], Optional[int], Optional[int], Optional[int], Optional[int], Optional[int], Optional[int]) -> str
```

Comment:
By default mypy uses ["implicit optional"](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-optional), i.e. marking an argument as `Optional` is not required when the default is `None`. So we can simplify:

```suggestion
    # type: (int, int, int, int, int, int, int, int) -> str
```